src/sdk/pynni/nni/smac_tuner/convert_ss_to_scenario.py

```diff
@@ -0,0 +1,119 @@
+    # parameter_name ordinal {value_1, ..., value_N} [default value]
+    # parameter_name integer [min_value, max_value] [default value]
+    # parameter_name integer [min_value, max_value] [default value] log
+    # parameter_name real [min_value, max_value] [default value]
+    # parameter_name real [min_value, max_value] [default value] log
+    # https://automl.github.io/SMAC3/stable/options.html
+    search_space = nni_search_space_content
+    with open('param_config_space.pcs', 'w') as pcs_fd:
+        if isinstance(search_space, dict):
+            for key in search_space.keys():
+                if isinstance(search_space[key], dict):
+                    try:
+                        if search_space[key]['_type'] == 'choice':
+                            pcs_fd.write('%s categorical {%s} [%s]\n' % (
+                                key, 
+                                str(search_space[key]['_value'])[1:-1], 
+                                search_space[key]['_value'][0]))
+                        elif search_space[key]['_type'] == 'randint':
+                            # TODO: support lower bound in randint
+                            pcs_fd.write('%s integer [0, %d] [%d]\n' % (
+                                key, 
+                                search_space[key]['_value'][0], 
+                                search_space[key]['_value'][0]))
+                        elif search_space[key]['_type'] == 'uniform':
+                            pcs_fd.write('%s real %s [%f]\n' % (
+                                key, 
+                                search_space[key]['_value'], 
+                                search_space[key]['_value'][0]))
+                        elif search_space[key]['_type'] == 'loguniform':
+                            pcs_fd.write('%s real %s [%f] log\n' % (
+                                key, 
+                                search_space[key]['_value'], 
+                                search_space[key]['_value'][0]))
+                        elif search_space[key]['_type'] == 'quniform' \
+                            and search_space[key]['_value'][2] == 1:
+                            pcs_fd.write('%s integer [%d, %d] [%d]\n' % (
+                                key,
+                                search_space[key]['_value'][0],
+                                search_space[key]['_value'][1],
+                                search_space[key]['_value'][0]))
+                        else:
+                            raise RuntimeError('unsupported _type %s' % search_space[key]['_type'])
+                    except:
+                        raise RuntimeError('_type or _value error.')
+        else:
+            raise RuntimeError('incorrect search space.')
+
+def generate_scenario(ss_content):
+    # deterministic, 1/0
+    # output_dir,
+    # paramfile, 
+    # run_obj, 'quality'
+    sce_fd = open('scenario.txt', 'w')
```

Comment:
could you use with open XXX to instead? ( for same code style with other code...

Reply:
done